File: util/android/src/main/java/com/dmi/util/AndroidPaths.java
package com.dmi.util;

import android.content.Context;

import java.io.File;

import static android.os.Environment.MEDIA_MOUNTED;
import static android.os.Environment.getExternalStorageState;
import static android.os.Environment.isExternalStorageRemovable;

public abstract class AndroidPaths {
    public static File getCachedDir(Context context) {
        return MEDIA_MOUNTED.equals(getExternalStorageState()) || !isExternalStorageRemovable() ?
               context.getExternalCacheDir() :
               context.getCacheDir();
    }
}

File: util/android/src/main/kotlin/com/dmi/util/AndroidPaths.kt
package com.dmi.util

import android.content.Context
import android.os.Environment.*
import java.io.File

object AndroidPaths {
    fun getCachedDir(context: Context): File {
        return if (MEDIA_MOUNTED == getExternalStorageState() || !isExternalStorageRemovable())
            context.externalCacheDir
        else
            context.cacheDir
    }
}
